.github/workflows/build-wheels.yml

```diff
@@ -67,36 +67,192 @@ jobs:
 
       - name: build manylinux with rust docker image
         if: matrix.os == 'ubuntu-22.04'
-        run: docker build -t rustc-manylinux2014_${{ matrix.cibw_arch }} python/scripts/rustc-manylinux2014_${{ matrix.cibw_arch }}
+        run: docker build -t rustc-manylinux2014_${{ matrix.cibw-arch }} python/scripts/rustc-manylinux2014_${{ matrix.cibw-arch }}
 
       - name: build metatensor-core wheel
         run: python -m cibuildwheel python/metatensor-core
         env:
           CIBW_BUILD: cp311-*
           CIBW_SKIP: "*musllinux*"
-          CIBW_ARCHS: ${{ matrix.cibw_arch }}
+          CIBW_ARCHS: ${{ matrix.cibw-arch }}
           CIBW_BUILD_VERBOSITY: 2
           CIBW_MANYLINUX_X86_64_IMAGE: rustc-manylinux2014_x86_64
           CIBW_MANYLINUX_AARCH64_IMAGE: rustc-manylinux2014_aarch64
+          CIBW_ENVIRONMENT: >
+            METATENSOR_NO_LOCAL_DEPS=1
+            MACOSX_DEPLOYMENT_TARGET=11
 
       - name: check wheels with twine
         run: twine check wheelhouse/*
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: wheels
+          name: core-wheel-${{ matrix.os }}-${{ matrix.cibw-arch }}
           path: ./wheelhouse/*.whl
 
-      - name: upload wheel to GitHub release
-        if: startsWith(github.ref, 'refs/tags/metatensor-core-v')
-        uses: softprops/action-gh-release@v1
+  build-torch-wheels:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.name }} (torch v${{ matrix.torch-version }})
+    strategy:
+      matrix:
+        torch-version: ['1.11', '1.12', '1.13', '2.0', '2.1', '2.2']
+        arch: ['arm64', 'x86_64']
+        os: ['ubuntu-22.04', 'macos-11', 'macos-14', 'windows-2019']
```

Comment:
Shouldn't there be another loop over python versions?

Reply:
no, we only need to build the code once per torch version thanks to https://github.com/lab-cosmo/metatensor/pull/564

We just pick one python version that's compatible with the torch version for the build below.

Reply:
I see.